manifests/mod/disk_cache.pp

```diff
@@ -17,44 +16,24 @@
 #   You can then control this behaviour in individual vhosts by explicitly defining CacheEnable.
 #
 # @note
-#   On Apache 2.4, mod_cache_disk installed.
+#   Apache 2.2, mod_disk_cache installed. On Apache 2.4, mod_cache_disk installed.
+#   This class is deprecated, use mode_cache_disk instead
 #
-# @see https://httpd.apache.org/docs/2.4/mod/mod_cache_disk.html for additional documentation.
+# @see https://httpd.apache.org/docs/2.4/mod/mod_cache_disk.html for additional documentation on version 2.4.
 #
 class apache::mod::disk_cache (
   Optional[Stdlib::Absolutepath] $cache_root = undef,
   Optional[String] $cache_ignore_headers     = undef,
   Boolean $default_cache_enable              = true,
 ) {
-  include apache
-  if $cache_root {
-    $_cache_root = $cache_root
-  } else {
-    $_cache_root = $facts['os']['family'] ? {
-      'Debian'  => '/var/cache/apache2/mod_cache_disk',
-      'RedHat'  => '/var/cache/httpd/proxy',
-      'FreeBSD' => '/var/cache/mod_cache_disk',
-    }
-  }
-
-  apache::mod { 'cache_disk': }
-
-  Class['apache::mod::cache'] -> Class['apache::mod::disk_cache']
-
-  $parameters = {
-    'default_cache_enable'  => $default_cache_enable,
-    '_cache_root'           => $_cache_root,
-    'cache_ignore_headers'  => $cache_ignore_headers,
-  }
+  deprecation('apache::mod::disk_cache', 'This class is deprecated; please use apache::mod::cache_disk')
 
-  # Template uses $_cache_root
-  file { 'disk_cache.conf':
-    ensure  => file,
-    path    => "${apache::mod_dir}/disk_cache.conf",
-    mode    => $apache::file_mode,
-    content => epp('apache/mod/disk_cache.conf.epp', $parameters),
-    require => Exec["mkdir ${apache::mod_dir}"],
-    before  => File[$apache::mod_dir],
-    notify  => Class['apache::service'],
+  class { 'apache::mod::cache_disk':
+    cache_root              => $cache_root,
+    cache_enable            => ['/'],
+    cache_ignore_headers    => $cache_ignore_headers,
```

Comment:
For backwards compatibility we may need something like:

```suggestion
    cache_ignore_headers    => $cache_ignore_headers.split(' '),
```

(I don't think it's worth breaking backwards compatibility to improve a legacy interface)